### PR TITLE
Re-organize the README

### PR DESCRIPTION
**Pull request by JST5000**
## High Level Overview of Change

Re-organizing the README to create a better experience for new developers & include more key links.

### Context of Change

We've been adding a lot of custom steps for React projects which have caused the README to be a bit cluttered. This cleans that up, and re-organizes for flow.

### Type of Change

- [X] Documentation Updates

## Before / After

* Kept the header the same to avoid breaking people's muscle memory on clicking the "Reference Doc" link
* Rephrased our features to be more in line with how the library is used
* Added more specific links to xrpl.org documentation & when to use it
* Moved the setup instructions for React & Deno into a separate linked file 
* Added links to the XRPL Developer Discord for help when developing
* Moved the Applications & Projects link to the bottom to focus on concrete code links near the top

### CODE REVIEW

**Comment by JST5000** [comment]
The diff is a bit of a mess, so it's helpful to directly read the doc directly: https://github.com/XRPLF/xrpl.js/blob/update-readme/README.md

File: README.md [hunk 1]
```diff
@@ -11,25 +11,25 @@ This is the recommended library for integrating a JavaScript/TypeScript app with
...
 
-What is `xrpl.js` used for? The applications on the list linked above use `xrpl.js`. Open a PR to add your app or project to the list!
+1. Managing keys & creating test credentials ([`Wallet`](https://js.xrpl.org/classes/Wallet.html) && [`Client.fundWallet()`](https://js.xrpl.org/classes/Client.html#fundWallet))
+2. Submitting transactions to the XRP Ledger ([`Client.submit(...)`](https://js.xrpl.org/classes/Client.html#submit) & [transaction types](https://xrpl.org/transaction-types.html))
+3. Sending requests to observe the ledger ([`Client.request(...)`](https://js.xrpl.org/classes/Client.html#request) using [public api methods](https://xrpl.org/public-api-methods.html))
+4. Subscribing to changes in the ledger ([Ex. ledger, transactions, & more...](https://xrpl.org/subscribe.html))
+5. Helper functions to parse ledger data (Ex. [`xrpToDrops`](https://js.xrpl.org/functions/xrpToDrops.html))
 
-### Features
+All of which works in Node.js (tested for v14+) & web browsers.
```
Comment (justinr1234) [inline README.md:22]: @JST5000 we only test chrome

**Review by justinr1234 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -11,25 +11,25 @@ This is the recommended library for integrating a JavaScript/TypeScript app with
 
 See the full reference documentation for all classes, methods, and utilities.
 
-## [➡️ Applications and Projects](https://github.com/XRPLF/xrpl.js/blob/main/APPLICATIONS.md)
+## Features
 
-What is `xrpl.js` used for? The applications on the list linked above use `xrpl.js`. Open a PR to add your app or project to the list!
+1. Managing keys & creating test credentials ([`Wallet`](https://js.xrpl.org/classes/Wallet.html) && [`Client.fundWallet()`](https://js.xrpl.org/classes/Client.html#fundWallet))
+2. Submitting transactions to the XRP Ledger ([`Client.submit(...)`](https://js.xrpl.org/classes/Client.html#submit) & [transaction types](https://xrpl.org/transaction-types.html))
+3. Sending requests to observe the ledger ([`Client.request(...)`](https://js.xrpl.org/classes/Client.html#request) using [public api methods](https://xrpl.org/public-api-methods.html))
```
Comment (mDuo13) [inline README.md:18]:
```suggestion
3. Sending requests to observe the ledger ([`Client.request(...)`](https://js.xrpl.org/classes/Client.html#request) using [public API methods](https://xrpl.org/public-api-methods.html))
```

File: README.md [hunk 3]
```diff
@@ -11,25 +11,25 @@ This is the recommended library for integrating a JavaScript/TypeScript app with
...
+4. Subscribing to changes in the ledger ([Ex. ledger, transactions, & more...](https://xrpl.org/subscribe.html))
+5. Helper functions to parse ledger data (Ex. [`xrpToDrops`](https://js.xrpl.org/functions/xrpToDrops.html))
```
Comment (mDuo13) [inline README.md:20]: Suggested changes for parallel structure and to add another example:

```suggestion
5. Parsing ledger data into more convenient formats ([`xrpToDrops`](https://js.xrpl.org/functions/xrpToDrops.html) and [`rippleTimeToISOTime`](https://js.xrpl.org/functions/rippleTimeToISOTime.html))
```

**Review by mDuo13 — APPROVED** [review]
Suggested a couple small improvements but overall this looks like a nice improvement.

**Comment by intelliot** [comment]
👍  for Rome's suggestions

**Pull request merged**